pkg/internal/cnsoperator/cnsfilevolumeclient/cnsfilevolumeclient.go

```diff
@@ -159,6 +165,33 @@ func (f *fileVolumeClient) AddClientVMToIPList(ctx context.Context, fileVolumeNa
 			return nil
 		}
 	}
+
+	// Prepare the CnsVolumeACLConfigureSpec
+	cnsVolumeID := cnstypes.CnsVolumeId{
+		Id: volumeID,
+	}
+	vSanFileShareNetPermissions := make([]vsanfstypes.VsanFileShareNetPermission, 0)
+	vsanFileShareAccessType := vsanfstypes.VsanFileShareAccessTypeREAD_WRITE
+	vSanFileShareNetPermissions = append(vSanFileShareNetPermissions, vsanfstypes.VsanFileShareNetPermission{
+		Ips:         clientVMIP,
+		Permissions: vsanFileShareAccessType,
+	})
+
+	cnsNFSAccessControlSpecList := make([]cnstypes.CnsNFSAccessControlSpec, 0)
+	cnsNFSAccessControlSpecList = append(cnsNFSAccessControlSpecList, cnstypes.CnsNFSAccessControlSpec{
+		Permission: vSanFileShareNetPermissions,
+	})
+
+	cnsVolumeACLConfigSpec := cnstypes.CnsVolumeACLConfigureSpec{
+		VolumeId:              cnsVolumeID,
+		AccessControlSpecList: cnsNFSAccessControlSpecList,
+	}
+	err = f.volumeManager.ConfigureVolumeACLs(ctx, cnsVolumeACLConfigSpec)
```

Comment:
I'm not sure why this was moved to `AddClientVMToIPList`. The `FileVolumeClient` interface is just needed to maintain the internal mapping of IPs -> VMs for a given file volume. I think configuring of the CNS file share should be done by the caller that wants to maintain this mapping

Reply:
IMO it is good to have all the operations in a single lock at one place, since the design mentions about serialization of the steps:
1. Get ExternalIPtoClientVms[IP] list from FileVolumeClients instance for SV-PVC.
2. Call CNS API to configure these IPAddresses on file volume.
3. Call AddClientVmToIPList() to add CnsFileAccessConfig.Spec.VmName

We can change the name of the method `AddClientVMToIPList` to indicate the correct behavior.
Let me know your thoughts on this

Reply:
> We can change the name of the method AddClientVMToIPList to indicate the correct behavior.

This interface was designed to only deal with `CnsFileVolumeClient` instances so i dont think its as simple as changing the name. 

Also, according to the design, we also configure CNS if the instance doesnt exist on the API server. That situation would need to be handled too. 

I think its better to keep the logic for these operations separate from each other

Reply:
That's reasonable. Thanks for helping me here to understand better. I have addressed your comment by keeping the configure CNS call in the cnsfileaccessconfig controller. Also, I have updated the test results in the MR description.

Also, I would like add the locks per IP after this PR is merged